rcl/test/rcl/test_graph.cpp

```diff
@@ -310,7 +313,218 @@ check_graph_state(
   }
 }
 
-/* Test graph queries with a hand crafted graph.
+typedef std::function<rcl_ret_t (const rcl_node_t *,
+  const char * node_name,
+  rcl_names_and_types_t *)> GetTopicsFunc;
+
+void expect_topics_types(const rcl_node_t * node, GetTopicsFunc &func, size_t num_topics, const char * topic_name) {
+  rcl_ret_t ret;
+  rcl_names_and_types_t nat{};
+  nat = rcl_get_zero_initialized_names_and_types();
+  ret = func(node, topic_name, &nat);
+  EXPECT_EQ(num_topics, nat.names.size);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  rcl_reset_error();
+  ret = rcl_names_and_types_fini(&nat);
+  rcl_reset_error();
+}
+
+/**
+ * This test creates publishers, subscribers, and services, then calls node graph functions
+ * from the perspective of each node.
+ * It verifies each node perceives the same graph.
+ */
+TEST_F(CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION), test_node_info_functions) {
```

Comment:
I think it is a good idea to break this test up to help with readability. At the very least we can have a test for each function; common setup/teardown can be put into a new test fixture, extending `TestGraphFixture`, if needed.

Reply:
Will do